File: backend/hackathon/src/main/java/com/dev_trails/hackathon/BillingTransaction.java
package com.dev_trails.hackathon;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;
@Entity
public class BillingTransaction {
@Id
@GeneratedValue(strategy=GenerationType.IDENTITY)
public Long id;
@Column(unique=true)
public String txnId;
public String type;
public String riderName;
public Double amount;
public String date;
public String description;
public String policyRef;
}

File: backend/models/BillingTransaction.js
const mongoose = require('mongoose');

const BillingTransactionSchema = new mongoose.Schema({
  txn_id: { type: String, unique: true, sparse: true },
  type: { type: String },
  rider_name: { type: String },
  amount: { type: Number },
  date: { type: String },
  description: { type: String },
  policy_ref: { type: String },
}, { timestamps: false });

BillingTransactionSchema.set('toJSON', { virtuals: true });
BillingTransactionSchema.virtual('id').get(function () { return this._id.toHexString(); });

module.exports = mongoose.model('BillingTransaction', BillingTransactionSchema);
